treeherder/perf/models.py

```diff
@@ -183,18 +172,20 @@ def _delete_in_chunks(self, repository, max_timestamp, chunk_size, sleep_time, l
                         break  # finished removing all expired data
                     else:
                         any_succesful_attempt = True
-                        logger.warning('Successfully deleted some rows')
-
-            if sleep_time:
-                # Allow some time for other queries to get through
-                time.sleep(sleep_time)
+                        logger.warning('Successfully deleted {} performance datum rows'.format(deleted_rows))
 
-    def _determine_max_timestamp(self, repo_name, cycle_interval, unsheriffed_repos_expire_days):
-        if repo_name in self.MAIN_REPOS:
-            return datetime.datetime.now() - cycle_interval
+    def _compute_ideal_chunk_size(self, max_timestamp, max_chunk_size):
```

Comment:
I'm wondering why calculate the length of the older data versus just using the max_chunk_size?

Reply:
Probably from my concern for the case when the number of records left to delete is less than the chunk size: The query will continue to with the table scan looking for more, and it will timeout doing so.  

The reason this is faster is because it is only an approximation: there will be some records with higher ids, but lower timestamps that could be deleted.  I assume that is relatively small

Reply:
I share Kyle's opinion on this.

Reply:
Ok, makes sense.